sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -0,0 +1,251 @@
+    this.children.add(span);
+    return span;
+  }
+
+  @Override
+  public @NotNull ISpan startChild(final @NotNull String operation) {
+    return root.startChild(operation);
+  }
+
+  @Override
+  public @NotNull ISpan startChild(
+      final @NotNull String operation, final @Nullable String description) {
+    return root.startChild(operation, description);
+  }
+
+  @Override
+  public @NotNull SentryTraceHeader toSentryTrace() {
+    return root.toSentryTrace();
+  }
+
+  @Override
+  public boolean finish() {
+    return this.finish(this.getStatus());
+  }
+
+  @Override
+  public boolean finish(@Nullable SpanStatus status) {
+    if (root.finish(status)) {
+      hub.withScope(
+          scope -> {
+            scope.withTransaction(
+                transaction -> {
+                  if (transaction == this) {
+                    scope.clearTransaction();
+                  }
+                });
+          });
+      SentryTransaction transaction = new SentryTransaction(this);
+      hub.captureTransaction(transaction);
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public void setOperation(final @NotNull String operation) {
+    this.root.setOperation(operation);
+  }
+
+  @Override
+  public @NotNull String getOperation() {
+    return this.root.getOperation();
+  }
+
+  @Override
+  public void setDescription(final @Nullable String description) {
+    this.root.setDescription(description);
+  }
+
+  @Override
+  public @Nullable String getDescription() {
+    return this.root.getDescription();
+  }
+
+  @Override
+  public void setStatus(final @Nullable SpanStatus status) {
+    this.root.setStatus(status);
+  }
+
+  @Override
+  public @Nullable SpanStatus getStatus() {
+    return this.root.getStatus();
+  }
+
+  @Override
+  public void setThrowable(final @Nullable Throwable throwable) {
+    this.root.setThrowable(throwable);
+  }
+
+  @Override
+  public @Nullable Throwable getThrowable() {
+    return this.root.getThrowable();
+  }
+
+  @Override
+  public @NotNull SpanContext getSpanContext() {
+    return this.root.getSpanContext();
+  }
+
+  @Override
+  public void setTag(final @NotNull String key, final @NotNull String value) {
+    this.root.setTag(key, value);
+  }
+
+  @Override
+  public @Nullable String getTag(final @NotNull String key) {
+    return this.root.getTag(key);
+  }
+
+  @Override
+  public boolean isFinished() {
+    return this.root.isFinished();
+  }
+
+  @Override
+  public @Nullable Boolean isSampled() {
+    return this.root.isSampled();
+  }
+
+  @Override
+  public void setName(@NotNull String name) {
+    this.root.setTag(ISpan.NAME_TAG, name);
+  }
+
+  @Override
+  public @NotNull String getName() {
+    return this.root.getTag(ISpan.NAME_TAG);
+  }
+
+  @Override
+  @Deprecated
+  @ApiStatus.ScheduledForRemoval
+  public void setRequest(@Nullable Request request) {
+    hub.configureScope(scope -> scope.setRequest(request));
```

Comment:
This is not great - ideally we would remove these methods already as it may be surprising to set request on a transaction and realise that it's set on the scope.

Reply:
I wonder if there are 2 transactions running at the same time and both call `setRequest`, they are gonna overwrite each other (at least on Android)

Reply:
We could just have a `request` field for now and when you do `var transation = new Transation` you set the fields into that. Same for anything else we need to call into the scope like this.

The deprecation note is there so later we can drop the code on next major.
Would this work?

Reply:
Yep, done that.